tools/cleanupPolicies.sh

```diff
@@ -4,13 +4,12 @@
 # Require a project id
 : ${1?"Usage: $0 projectid"}
 projectid=$1
-# get the policy bindings for the project
-bindings=$(gcloud projects get-iam-policy $projectid --format=json)
-# get the members of the BigQuery Job User role
-members=$(echo $bindings | jq '.bindings[] | if .role == "roles/bigquery.jobUser" then .members else empty end')
-# Loop through the members that start with "group:policy-" That is the signature of a SAM group. And I hope nothing else important!
-# Remove the members one by one - this is noisy, but leaving it that way for now so we see the results in the log
-for row in $(echo $members | jq -r '.[] | select(startswith("group:policy-"))'); do
-    echo "removing member: ${row}"
-    gcloud projects remove-iam-policy-binding $projectid --member=$row --role=roles/bigquery.jobUser > /dev/null
-done
+# retrieve all IAM policies for data project
+BINDINGS=$(gcloud projects get-iam-policy ${projectid} --format=json)
+# remove any policies that start with group:policy- or deleted:group:policy-
+# group policies are created as a part of our test run and need to be cleared out
+# to avoid hitting 250 IAM policy limit
+OK_BINDINGS=$(echo ${BINDINGS} | jq 'del(.bindings[] | select(.role=="roles/bigquery.jobUser") | .members[] | select(startswith("group:policy-") or startswith("deleted:group:policy-")))')
```

Comment:
That is quite a jq expression! Perhaps a comment to explain what it is doing would be useful for a future maintainer.

Reply:
JQ expression is courtesy of @fboulnois !

I added more detailed comments - let me know if these make sense, and @fboulnois - please free to edit.